src/Tools/dotnet-monitor/CollectionRules/Configuration/CollectionRuleConfigureNamedOptions.cs

```diff
@@ -57,14 +57,28 @@ private void BindActionSettings(IConfigurationSection ruleSection, CollectionRul
             if (null != actionOptions &&
                 _actionOperations.TryCreateOptions(actionOptions.Type, out object actionSettings))
             {
-                IConfigurationSection settingsSection = ruleSection.GetSection(ConfigurationPath.Combine(
-                    nameof(CollectionRuleOptions.Actions),
-                    actionIndex.ToString(CultureInfo.InvariantCulture),
-                    nameof(CollectionRuleActionOptions.Settings)));
+                int configurationIndex = 0;
+                int actionsFound = 0;
+
+                while (actionsFound <= actionIndex)
```

Comment:
I am not really understanding this logic. Why would there be action elements that are empty?

Reply:
This logic is checking for whether the current action is a custom shortcut or a normal action. A custom shortcut will have a value set, whereas for a standard action the value will be empty, but it'll have children elements describing the different properties. So, when we check `actionSection.Exists() && string.IsNullOrEmpty(actionSection.Value)`, that's basically saying: is there anything in this section, and is it populated by children, or directly by a value. If it's populated by children, then we know it's a normal action, in which case we do the binding. If it's populated by a value (aka "MyCustomShortcut"), then we leave it alone, and we handle it in `CustomShortcutsConfigureNamedOptions`. I tried to mostly separate the new logic from the old, but maybe it's more confusing having these two separate files for configuring options.

Reply:
I'm about to put up a cleaned-up version of this (and the other file) that uses your idea with the children - you can let me know if that feels any better to you, and then we can go from there.